src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,42 +1,70 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import { Loader } from '../Loader';
+import { Todo } from '../../types/Todo';
+import { getUser } from '../../api';
+import { User } from '../../types/User';
+
+type Props = {
+  todo: Todo;
+  onClose: () => void;
+};
+
+export const TodoModal: React.FC<Props> = ({ todo, onClose }) => {
+  const [selectedUser, setSelectedUser] = useState<User | null>(null);
+  const [loading, setLoading] = useState(false);
+
+  useEffect(() => {
+    setLoading(true);
+
+    getUser(todo.userId)
+      .then(setSelectedUser)
+      .finally(() => setLoading(false));
+  }, [todo.userId]);
 
-export const TodoModal: React.FC = () => {
   return (
     <div className="modal is-active" data-cy="modal">
       <div className="modal-background" />
 
-      <Loader />
+      {loading || !selectedUser ? (
+        <Loader />
```

Comment:
```suggestion
      {loading ? (
        <Loader />
```
You should show the loader only in case you are loading something